common/scala/src/main/scala/whisk/core/containerpool/ContainerFactory.scala

```diff
@@ -50,6 +50,15 @@ trait ContainerFactory {
   def cleanup(): Unit
 }
 
+object ContainerFactory {
+
+  /** include the instance name, if specified and strip invalid chars before attempting to use them in the container name */
+  def containerNamePrefix(instanceId: InstanceId): String =
+    s"wsk${instanceId.name.getOrElse("")}${instanceId.toInt}"
+      .replaceAll("[^a-zA-Z0-9_\\.\\-]", "") // based on https://github.com/moby/moby/issues/3138 and https://github.com/moby/moby/blob/master/daemon/names/names.go
```

Comment:
Since there’s a replace all subsequently should we remove this and trust the invariant is asserted elsewhere?

Reply:
Ah there’s a docker clean which would need this.